modules/chimera/src/main/java/org/dcache/chimera/JdbcFs.java

```diff
@@ -725,36 +719,40 @@ public FsInode path2inode(String path, FsInode startFrom) throws ChimeraFsExcept
     @Override
     public String inode2id(FsInode inode) throws ChimeraFsException {
         try {
-            return _idCache.get(inode.ino(), () -> {
+            return _idCache.get(inode.ino(), (key) -> {
                 String id = _sqlDriver.getId(inode);
                 if (id == null) {
-                    throw FileNotFoundChimeraFsException.of(inode);
+                    throw new RuntimeException(FileNotFoundChimeraFsException.of(inode));
```

Comment:
this looks quite dirty....

Reply:
The issue if I recall, is that you can't throw an exception in a CompletableFuture like you can in a ListenableFuture. Perhaps we can find a nicer way to propagate the FileNotFoundException upwards , I am almost sure, there is a nicer way to solve this issue, I just have not found it yet.

Reply:
This should be the case now.